Doppler.PushContact/Services/PushContactService.cs

```diff
@@ -231,6 +231,40 @@ public async Task AddHistoryEventsAsync(IEnumerable<PushContactHistoryEvent> pus
             }
         }
 
+        public async Task<IEnumerable<string>> GetAllDeviceTokensByDomainAsync(string domain)
+        {
+            if (string.IsNullOrEmpty(domain))
+            {
+                throw new ArgumentException($"'{nameof(domain)}' cannot be null or empty.", nameof(domain));
+            }
+
+            var FilterBuilder = Builders<BsonDocument>.Filter;
+
+            var filter = FilterBuilder.Eq(PushContactDocumentProps.DomainPropName, domain)
+                & FilterBuilder.Eq(PushContactDocumentProps.DeletedPropName, false);
+
+            var options = new FindOptions<BsonDocument>
+            {
+                Projection = Builders<BsonDocument>.Projection
+                .Include(PushContactDocumentProps.DeviceTokenPropName)
+                .Exclude(PushContactDocumentProps.IdPropName)
+            };
+
+            try
+            {
+                var pushContactsFiltered = await (await PushContacts.FindAsync(filter, options)).ToListAsync();
+
+                return pushContactsFiltered
+                    .Select(x => x.GetValue(PushContactDocumentProps.DeviceTokenPropName).AsString);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Error getting {nameof(PushContactModel)}s by {nameof(domain)} {domain}");
```

Comment:
@elsupergomez always recommends structured logging

```suggestion
                _logger.LogError(ex, "Error getting models by domain {domain}", domain);
```

Reply:
I am using string interpolation in all the solution but I will keep it in mind for a future refactor. Thanks!

Reply:
Agregué una tarea para que no se pierda esta sugerencia: https://makingsense.atlassian.net/browse/DOM-137. 
CC @hrocca